mrp_subcontracting_picking/models/mrp_production.py

```diff
@@ -0,0 +1,27 @@
+from odoo import api, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    @api.depends(
+        "move_raw_ids.state",
+        "move_raw_ids.quantity_done",
+        "move_finished_ids.state",
+        "workorder_ids",
+        "workorder_ids.state",
+        "product_qty",
+        "qty_producing",
+    )
+    def _compute_state(self):
+        super(MrpProduction, self)._compute_state()
+        for prod in self.filtered(lambda prod: prod.state == "done"):
+            pickings = prod.move_finished_ids.mapped(
+                "move_dest_ids.picking_id"
+            ).filtered(lambda p: p.state == "subcontracted")
+            if not pickings:
+                continue
+            for picking in pickings:
```

Comment:
Why do we need an iteration here?
[action_assign](https://github.com/odoo/odoo/blob/1761d1de485382a5c5dc2d5f421c315ce34e1a3e/addons/stock/models/stock_picking.py#L750) and [action_confirm](https://github.com/odoo/odoo/blob/1761d1de485382a5c5dc2d5f421c315ce34e1a3e/addons/stock/models/stock_picking.py#L738) both work with recordsets

Reply:
Resolved